companion/src/modeledit/logicalswitches.cpp

```diff
@@ -208,75 +248,65 @@ void LogicalSwitchesPanel::delayEdited(double delay)
   }
 }
 
-void LogicalSwitchesPanel::edited()
+void LogicalSwitchesPanel::offsetEdited()
 {
-  if (!lock) {
-    lock = true;
-    int i = sender()->property("index").toInt();
-    CSFunctionFamily oldFuncFamily = model->logicalSw[i].getFunctionFamily();
-    model->logicalSw[i].func = csw[i]->itemData(csw[i]->currentIndex()).toInt();
-    CSFunctionFamily newFuncFamily = model->logicalSw[i].getFunctionFamily();
-
-    if (oldFuncFamily != newFuncFamily) {
-      model->logicalSw[i].clear();
-      model->logicalSw[i].func = csw[i]->currentData().toInt();
-      if (newFuncFamily == LS_FAMILY_TIMER) {
-        model->logicalSw[i].val1 = -119;
-        model->logicalSw[i].val2 = -119;
-      }
-      else if (newFuncFamily == LS_FAMILY_EDGE) {
-        model->logicalSw[i].val2 = -129;
-      }
-      setSwitchWidgetVisibility(i);
-    }
+  offsetEditedAt(sender()->property("index").toInt());
+}
 
-    RawSource source;
+bool LogicalSwitchesPanel::offsetEditedAt(int index)
+{
+  if (lock)
+    return false;
 
-    switch (newFuncFamily)
+  lock = true;
+  bool mod = false;
+  int value;
+
+  switch (model->logicalSw[index].getFunctionFamily())
+  {
+    case LS_FAMILY_VOFS:
     {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Extract this nested code block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1199)